lib/commands/performance.js

```diff
@@ -61,6 +73,59 @@ commands.getPerformanceData = async function getPerformanceData (packageName, da
   return data;
 };
 
+// API level 30 and above
+//   ['<System Type>', '<Memory Type>', <pss total>, <private dirty>, <private clean>, <swapPss dirty>, <rss total>, <heap size>, <heap alloc>, <heap free>]
+memoryParsers.Api30AndAbove = function Api30AndAbove (entries, valDict) {
+  const [type, subType] = entries;
+  if (type === MEMORY_CONSTANT.NATIVE && subType === MEMORY_CONSTANT.HEAP) {
+    [,, valDict.nativePss, valDict.nativePrivateDirty,,, valDict.nativeRss, valDict.nativeHeapSize, valDict.nativeHeapAllocatedSize] = entries;
+  } else if (type === MEMORY_CONSTANT.DALVIK && subType === MEMORY_CONSTANT.HEAP) {
+    [,, valDict.dalvikPss, valDict.dalvikPrivateDirty,,, valDict.dalvikRss] = entries;
+  } else if (type === MEMORY_CONSTANT.EGL && subType === MEMORY_CONSTANT.MTRACK) {
+    [,, valDict.eglPss, valDict.eglPrivateDirty] = entries;
+  } else if (type === MEMORY_CONSTANT.GL && subType === MEMORY_CONSTANT.MTRACK) {
+    [,, valDict.glPss, valDict.glPrivateDirty] = entries;
+  } else if (type === MEMORY_CONSTANT.TOTAL && entries.length === 9) {
+    // has 9 entries
+    [, valDict.totalPss, valDict.totalPrivateDirty,,, valDict.totalRss] = entries;
+  }
+};
+
+// API level between 18 and 30
+//   ['<System Type>', '<Memory Type>', <pss total>, <private dirty>, <private clean>, <swapPss dirty>, <heap size>, <heap alloc>, <heap free>]
+// except 'TOTAL', which skips the second type name
+memoryParsers.Api18And30 = function Api18And30 (entries, valDict) {
```

Comment:
I'd rather return valDict than mutate it in-place

Reply:
The current solution is also ok, but either the code or the method documentation must explicitly state that we intentionally mutate the incoming argument